plaidml/bridge/openvino/ops/deformable_convolution.cpp

```diff
@@ -0,0 +1,389 @@
+                    .input_layout(plaidml::op::TensorLayout::NCX)
+                    .filter_layout(plaidml::op::TensorLayout::KCX)
+                    .autogroup_mode(plaidml::op::AutoGroupMode::EXPLICIT)
+                    .group_layout(plaidml::op::GroupLayout::IN_C)
+                    .group_layout(plaidml::op::GroupLayout::IN_K)
+                    .groups(G);
+  return result;
+}
+
+// Compute 3D DeformableConvolution.
+edsl::Tensor compute_3d_deformable_convolution(edsl::Tensor I, edsl::Tensor OFF, edsl::Tensor F,
+                                               std::vector<int64_t> I_shape, std::vector<int64_t> OFF_shape,
+                                               std::vector<int64_t> F_shape, int G, int DG, size_t rank,
+                                               std::vector<size_t> strides, std::vector<size_t> dilations,
+                                               std::vector<size_t> pad_befores, plaidml::op::AutoPadMode autopad_mode) {
+  // Define dim of each tensor.
+  int N = I_shape[0];
+  int CI = I_shape[1];
+  int CO = F_shape[1];
+  int OFF_C = OFF_shape[1];
+  int D = I_shape[2];
+  int H = I_shape[3];
+  int W = I_shape[4];
+  int OFF_D = OFF_shape[2];
+  int OFF_H = OFF_shape[3];
+  int OFF_W = OFF_shape[4];
+  int F_D = F_shape[2];
+  int F_H = F_shape[3];
+  int F_W = F_shape[4];
+  // Throw exception
+  if (CI % G != 0 || CO % G != 0 || CI % DG != 0 || OFF_C % DG != 0) {
+    THROW_IE_EXCEPTION << "Incorrected shape for DeformableConvolution.";
+  }
+  // Define new depth, height and width.
+  int NEW_D = F_D * OFF_D;
+  int NEW_H = F_H * OFF_H;
+  int NEW_W = F_W * OFF_W;
+  // Get offset of depth, height and width.
+  edsl::Tensor offset = op::reshape(OFF, make_tuple<int64_t>({N, DG, 3 * F_D * F_H * F_W, OFF_D, OFF_H, OFF_W}));
+  edsl::Tensor offset_d = op::slice(offset)
+                              .add_dim(0, N)
+                              .add_dim(0, DG)
+                              .add_dim(0, OFF_C / DG - 2, 3)
+                              .add_dim(0, OFF_D)
+                              .add_dim(0, OFF_H)
+                              .add_dim(0, OFF_W);
+  edsl::Tensor offset_h = op::slice(offset)
+                              .add_dim(0, N)
+                              .add_dim(0, DG)
+                              .add_dim(1, OFF_C / DG - 1, 3)
+                              .add_dim(0, OFF_D)
+                              .add_dim(0, OFF_H)
+                              .add_dim(0, OFF_W);
+  edsl::Tensor offset_w = op::slice(offset)
+                              .add_dim(0, N)
+                              .add_dim(0, DG)
+                              .add_dim(2, OFF_C / DG, 3)
+                              .add_dim(0, OFF_D)
+                              .add_dim(0, OFF_H)
+                              .add_dim(0, OFF_W);
+  offset_d = op::reshape(offset_d, make_tuple<int64_t>({N, DG, F_D, F_H, F_W, OFF_D, OFF_H, OFF_W}));
```

Comment:
Can we reshape offset into something like
 ` auto offset_dhw = op::reshape(offset, make_tuple<int64_t>({N, DG, 3, F_D, F_H, F_W, OFF_D, OFF_H, OFF_W}));`
instead of 3 tensors for d, h, w separately?

Reply:
updated the shape for reshapeOp, '3' should be after `F_D, F_H, F_W,`

Reply:
ok, that make sense, i will fix it, maybe i will implement it in common deformableconvolution for multi-dimension

Reply:
That's good.